src/transforms/babel/babelrc.js

```diff
@@ -0,0 +1,219 @@
+const {buildRootChain} = require('@babel/core/lib/config/config-chain');
+const {mergeOptions} = require('@babel/core/lib/config/util');
+const semver = require('semver');
+const logger = require('../../logger');
+const path = require('path');
+const promisify = require('../../utils/promisify');
+const resolve = promisify(require('resolve'));
+const installPackage = require('../../utils/installPackage');
+
+async function getBabelConfig(asset, isSource) {
+  let config = await getBabelRc(asset, isSource);
+  if (!config) {
+    return null;
+  }
+
+  // Ignore if the config is empty.
+  if ((!config.plugins || config.plugins.length === 0) ||
+    (!config.presets || config.presets.length === 0)) {
+    return null;
+  }
+
+  let babelVersion = await getBabelVersion(asset, config);
+  
+  return {
+    babelVersion,
+    config
+  };
+}
+
+module.exports = getBabelConfig;
+
+/**
+ * Finds a .babelrc for an asset. By default, .babelrc files inside node_modules are not used.
+ * However, there are some exceptions:
+ *   - if `browserify.transforms` includes "babelify" in package.json (for legacy module compat)
+ *   - the `source` field in package.json is used by the resolver
+ */
+async function getBabelRc(asset, isSource) {
+  // Support legacy browserify packages
+  let pkg = await asset.getPackage();
+  let browserify = pkg && pkg.browserify;
+  if (browserify && Array.isArray(browserify.transform)) {
+    // Look for babelify in the browserify transform list
+    let babelify = browserify.transform.find(
+      t => (Array.isArray(t) ? t[0] : t) === 'babelify'
+    );
+
+    // If specified as an array, override the config with the one specified
+    if (Array.isArray(babelify) && babelify[1]) {
+      return babelify[1];
+    }
+
+    // Otherwise, return the .babelrc if babelify was found
+    return babelify ? await findBabelRc(asset) : null;
+  }
+
+  // If this asset is not in node_modules, always use the .babelrc
+  if (isSource) {
+    return await findBabelRc(asset);
+  }
+
+  // Otherwise, don't load .babelrc for node_modules.
+  // See https://github.com/parcel-bundler/parcel/issues/13.
+  return null;
+}
+
+async function findBabelRc(asset) {
+  // TODO: support babelignore, etc.
+  return await asset.getConfig(['.babelrc', '.babelrc.js'], {
```

Comment:
@loganfsmyth would be nice if babel could expose an API for loading babelrc that returned it raw instead of normalizing. Unfortunately, using `loadPartialConfig` returns a result that isn't compatible with babel 6. So I will need to reimplement support for babelignore, etc. myself, which isn't great...

Reply:
> Can we fork config resolution based on the Babel version?

Maybe... but we do programmatically modify the config in some cases, so it would be nice not to need to have multiple versions of that code.

> Also, .babelrc.js files can export functions FYI

ok, we'd have to support that too.

> does this load .babelrc files relative to the file being compiled?

Yes...